source/finite_element_field.cc

```diff
@@ -86,14 +87,13 @@ void FE_FieldBase<dim>::reinit()
   old_solution.reinit(solution);
   old_old_solution.reinit(solution);
 
+  // distributed vector contains only locally owned DoFs
   #ifdef USE_PETSC_LA
     distributed_vector.reinit(locally_owned_dofs,
                               mpi_communicator);
   #else
     distributed_vector.reinit(locally_owned_dofs,
-                              locally_relevant_dofs,
-                              mpi_communicator,
-                              true);
+                              mpi_communicator);
```

Comment:
Are you sure that both constructors share the same arguments? This is somewhat of a big change if not.

Reply:
I think this was an issue with PETSc. Shouldn't the `distributed_vector` only contain the `locally_owned_dofs`?

Reply:
Yes, the dimension of  `distributed_vector` has to correspond to the number of locally owned degrees of freedom. Considering that the constructor you replaced is the one we have been using so far, it already generates a vector with the correct dimension. Therefore, I am not sure if this constructors are equivalent and if not, this may may cause issues.

Reply:
In the MPI framework, ghosted values can also be stored in a vector. In the previous version, this was the case and because of the additional `true` argument, the ghosted values could be modified. Communication happens when calling the `compress` method. The current version does not allow to stores any ghosted value in a distributed vector.